src/Compilers/CSharp/Portable/Symbols/MergedNamespaceSymbol.cs

```diff
@@ -168,17 +168,42 @@ private ImmutableArray<Symbol> SlowGetChildrenOfName(ReadOnlyMemory<char> name)
         /// </summary>
         private HashSet<ReadOnlyMemory<char>> SlowGetChildNames(IEqualityComparer<ReadOnlyMemory<char>> comparer)
         {
-            var childNames = new HashSet<ReadOnlyMemory<char>>(comparer);
+            // compute the final capacity of the set we'll return, to reduce heap churn
+            int childCount = 0;
+            List<ImmutableArray<Symbol>> namespacesMembers = new(capacity: _namespacesToMerge.Length);
```

Comment:
>List [](http://example.com/codeflow?start=12&length=4)

Can we use an ```ArrayBuilder<ImmutableArray<Symbol>>``` instead? We usually get those from a pool, which allows reuse of instances.

Reply:
Agreed.  This will otherwise show up as a lot of garbage in this hotpath.